src/openassetio-core/src/hostApi/EntityReferencePager.cpp

```diff
@@ -21,6 +23,17 @@ EntityReferencePager::EntityReferencePager(
     managerApi::HostSessionPtr hostSession)
     : pagerInterface_(std::move(pagerInterface)), hostSession_(std::move(hostSession)) {}
 
+EntityReferencePager::~EntityReferencePager() {
+  try {
+    pagerInterface_->close(hostSession_);
+  } catch (const std::exception& ex) {
+    hostSession_->logger()->error(ex.what());
+  } catch (...) {
+    hostSession_->logger()->error(
+        "Unknown exception caught from EntityReferencePagerInterface::close");
```

Comment:
I'll go with that for consistency, although it's worth remembering that just cause something dosen't inherit from `std::exception` dosen't necessarily mean it's not an exception object. (Although, practically I expect 99% of things that are intended to be exceptions do in fact inherit from std::exception)

Reply:
https://github.com/OpenAssetIO/OpenAssetIO/pull/1013/commits/4f541a78dd1505ef737a45c2317def4a023b83c3

Reply:
> just cause something dosen't inherit from std::exception dosen't necessarily mean it's not an exception object

That's exactly the ambiguity I was referring to and trying to avoid, it's tricky to word succinctly though.  I'm pretty sure if a C++ dev sees "non-exception object caught" they'll understand what that means, so this LGTM.